Microsoft.Toolkit.Uwp/Helpers/BackgroundTaskHelper.cs

```diff
@@ -0,0 +1,183 @@
+        {
+            // Check if the task is already registered.
+            if (IsBackgroundTaskRegistered(backgroundTaskName) == true)
+            {
+                BackgroundTaskRegistration previouslyRegistered = GetBackgroundTask(backgroundTaskName) as BackgroundTaskRegistration;
+
+                if (forceRegister == true)
+                {
+                    Unregister(previouslyRegistered);
+                }
+                else
+                {
+                    return null;
+                }
+            }
+
+            // build the background task
+            BackgroundTaskBuilder builder = new BackgroundTaskBuilder();
+            builder.Name = backgroundTaskName;
+
+            // check if we are registering in SPM mode
+            if (backgroundTaskEntryPoint != string.Empty)
+            {
+                builder.TaskEntryPoint = backgroundTaskEntryPoint;
+            }
+
+            builder.CancelOnConditionLoss = enforceConditions;
+
+            // add the conditions if we have them.
+            conditions?.ToList().ForEach(builder.AddCondition);
+
+            builder.SetTrigger(trigger);
+
+            // Register it
+            BackgroundTaskRegistration registered = builder.Register();
+
+            return registered;
+        }
+
+        /// <summary>
+        /// Register a background task with conditions.
+        /// If the task is already registered and has the same trigger, returns the existing registration if it has the same trigger.
+        /// If the task is already registered but has different trigger, return null by default.
+        /// Or set <paramref name="forceRegister"/> to true to un-register the old one and then re-register.
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task. This class has to implement IBackgroundTask</param>
+        /// <param name="trigger">Trigger that indicate when the background task should be invoked</param>
+        /// <param name="forceRegister">Indicate if the background task will be force installed in the case of being already registered</param>
+        /// <param name="enforceConditions">Indicate if the background task should quit if condition is no longer valid</param>
+        /// <param name="conditions">Optional conditions for the background task to run with</param>
+        /// <returns>Background Task that was registered with the system</returns>
+        public static BackgroundTaskRegistration Register(Type backgroundTaskType, IBackgroundTrigger trigger, bool forceRegister = false, bool enforceConditions = true, params IBackgroundCondition[] conditions)
+        {
+            return Register(backgroundTaskType.Name, backgroundTaskType.FullName, trigger, forceRegister, enforceConditions, conditions);
```

Comment:
FullName is the full name with the class name and the namespace, which is what we want for the TaskEntryPoint. backgroundTaskName is just the name of the class.